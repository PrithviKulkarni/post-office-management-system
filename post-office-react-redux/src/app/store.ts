import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import letterReducer from '../features/LetterSlice';
import parcelReducer from '../features/ParcelSlice';
import depotReducer from '../features/DepotSlice';
import driverReducer from '../features/DriverSlice' 
export const store = configureStore({
  reducer: {
    letter: letterReducer,
    parcel: parcelReducer,
    depot: depotReducer,
    driver: driverReducer
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
