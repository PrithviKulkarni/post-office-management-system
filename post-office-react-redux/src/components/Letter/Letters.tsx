import { Card } from "react-bootstrap";
import { getAllLetters } from "../../features/LetterSlice";
import { Link } from "react-router-dom";
import { useAppSelector } from '../../app/hooks';

const Letters = () => {
  const letters = useAppSelector(getAllLetters);

  return (
    <div style={{
        alignItems: "center",
        display: "flex",
        flexDirection: "column",
      }}
      data-testid="letterList ">
      {letters.map((letter, idx) => (
        <Card
          style={{
            width: "25rem",
            padding: "1rem",
            margin: "5px",
            backgroundColor: "rgb(40,40,40)",
            display: "flex",
            justifyContent: "center",
            alignItems: "center",
          }}
          key={letter.id}
        >
          <Card.Body>
            <Card.Title style={{ color: "white", fontWeight: "bold" }}>
              {letter.address}
            </Card.Title>
            <Card.Subtitle className="mb-2 text-muted">
              {letter.dispatched.toString().slice(0, 24)}
            </Card.Subtitle>
            <Card.Text style={{ color: "white" }}>
              {letter.delivery_class}
            </Card.Text>
            <Card.Subtitle className="mb-2 text-muted">
              {letter.delivered ? "Delivered" : "Not Delivered" }
            </Card.Subtitle>
            <Link to={`/editLetter/${letter.id}`} data-testid="edit-button">Edit Letter</Link>
            <br />
            <Link to={`/deleteLetter/${letter.id}`} data-testid="delete-button">Delete Letter</Link>
          </Card.Body>
        </Card>
      ))}
    </div>
  );
};

export default Letters;
