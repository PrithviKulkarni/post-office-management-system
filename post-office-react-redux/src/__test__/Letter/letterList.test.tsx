import React from "react";
import { render, cleanup, screen } from "@testing-library/react";
import Letters from "../../components/Letter/Letters";
import { store } from "../../app/store";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

afterEach(cleanup);
it("renders Letters page correctly", () => {
  const { asFragment } = render(
    <Provider store={store}>
      <Letters />
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("Verifies the Letters render", () => {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <Letters />
      </BrowserRouter>
    </Provider>
  );

  expect(screen.getByTestId("letterList")).toBeInTheDocument();
});

