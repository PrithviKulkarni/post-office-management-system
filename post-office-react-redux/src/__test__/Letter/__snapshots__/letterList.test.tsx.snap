// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Letters page correctly 1`] = `
<DocumentFragment>
  <div
    data-testid="letterList "
    style="align-items: center; display: flex; flex-direction: column;"
  />
</DocumentFragment>
`;
